### ISSUE TITLE
Evaluation code for LIBERO

### ISSUE THREAD

**Issue by Cheny4ng** [declare-lab/nora#11]
Hi,

Thanks and congratulations on this excellent work! Just wondering if there’s any chance you could release the evaluation code used for Libero testing?

Really appreciate it—thanks in advance!

**Comment by Yysrc** [comment]
Hi, I've supplemented the Libero testing section. Here's the code, and I hope it will be helpful to you.https://github.com/[Yysrc/nora_libero](https://github.com/Yysrc/nora_libero)

**Comment by AndersonYu7** [comment]
**Issue: LIBERO Fails to Match Reported Paper Accuracy**

I have tested two different approaches, but neither achieves the accuracy reported in the LIBERO paper:

1. **Training on `nora-long` model** using the modified LIBERO-Object dataset provided by OpenVLA.  
   - **Accuracy achieved**: 84.0%

2. **Testing the pretrained model** `declare-lab/nora-long-finetuned-libero-object`.  
   - **Accuracy achieved**: Only 67.0%

Both tests were conducted using the official evaluation script from OpenVLA.

@Yysrc Could you share the results you've achieved when running these models?

**Comment by Cheny4ng** [comment]
> Hi, I've supplemented the Libero testing section. Here's the code, and I hope it will be helpful to you. https://github.com/Yysrc/nora_libero

Hi @AndersonYu7 , thank you so much for your code! Can I ask what unnorm_key you use for testing? I tried to use "libero_spatial" but it wasn't found in the model, and the default "bridge_orig"  unnorm_key just didn't work well.

**Comment by hungchiayu1** [comment]
@AndersonYu7 I have just reran using my libero eval for this checkpoint declare-lab/nora-long-finetuned-libero-object. I was able to replicate the original result in the paper 95.2% acc. Could you check [experiments/libero/run_libero_eval_nora.py](https://github.com/declare-lab/nora/blob/main/experiments/libero/run_libero_eval_nora.py) and see if you are able to replicate my result using this script?

**Comment by AndersonYu7** [comment]
@hungchiayu1 
I re-ran the evaluation using your script `experiments/libero/run_libero_eval_nora.py` on:
- the checkpoint `declare-lab/nora-long-finetuned-libero-object`
- my own fine-tuned checkpoint

I observed **67.2%** and **84.0%**, which are similar to the results I previously obtained. Full logs are attached for inspection.

[EVAL-libero_object-qwen-2025_08_02-20_00_22.txt](https://github.com/user-attachments/files/21569534/EVAL-libero_object-qwen-2025_08_02-20_00_22.txt)
[EVAL-libero_object-qwen-2025_08_03-18_31_57.txt](https://github.com/user-attachments/files/21569533/EVAL-libero_object-qwen-2025_08_03-18_31_57.txt)

The repository does not include `libero_utils.py` and `robot_utils.py`, so to get the script to run I copied those two modules from the OpenVLA-OFT and made some adjustments to make them compatible with this evaluation script. 

[libero_utils.txt](https://github.com/user-attachments/files/21569543/libero_utils.txt)
[robot_utils.txt](https://github.com/user-attachments/files/21569542/robot_utils.txt)
[run_libero_eval_nora.txt](https://github.com/user-attachments/files/21569544/run_libero_eval_nora.txt)

During the evaluation I also saw two non-fatal issues that might subtly affect the reported accuracy:
1. Repeated warning:  
  <img width="877" height="111" alt="Image" src="https://github.com/user-attachments/assets/6052d887-8e44-4340-ba4f-45b114cba110" />
  This indicates the tokenizer did not have an explicit pad token, so it fell back to using the EOS token. 


2. A detokenization-related problem.
<img width="1601" height="136" alt="Image" src="https://github.com/user-attachments/assets/d890a85b-4648-40f8-b12e-22fdd4cd1cdf" />


Although these issues don’t prevent the run from completing, could they be affecting the reported results? Is this a bug in the evaluation code, or caused by my configuration / the substitutions I made? Thanks!

**Comment by hungchiayu1** [comment]
Let me look into this. Give me awhile

**Comment by hungchiayu1** [comment]
@jiuyixu25 For your output it seems like u did not set the tokenizer time_horizon correctly. nora-finetuned-libero-spatial should generate an action horizon of 1 instead of 5, therefore you are getting this warning. To generate an action horizon of 5, you should use the nora-long-finetuned-libero-spatial checkout.

**Comment by AndersonYu7** [comment]
> Can I check if you are inferencing in bf16 and with flash-attn or sdpa

Ya, I’m inferencing in bf16 and not using flash-attn (SDPA only).

Regarding the Error decoding issue:
``` bash
Error decoding tokens: cannot reshape array of size 33 into shape (7)
Tokens: [271, 326, 340, 372, 271, 326, 1512, 1683, 297, 803, 333, 258]
```
Although it doesn’t crash the run, whenever this “error decoding” appears, that episode has a high chance of failing.
From [this Hugging Face discussion](https://huggingface.co/physical-intelligence/fast/discussions/4), it seems such decoding errors might result in the model returning an all-zero action.
Have you encountered the same issue on your side?

**Comment by AndersonYu7** [comment]
@hungchiayu1 Hi, Could you please provide the videos for the following task and episodes?

Task: pick up the alphabet soup and place it in the basket
Episodes: 3, 5, 7

I consistently observe failures in these episodes, so having the videos would help me better understand what’s going wrong.

Thanks a lot!

**Comment by soujanyaporia** [comment]
Hey everyone! We released NORA-1.5: https://huggingface.co/declare-lab/nora-1.5
